### DISCUSSION TITLE
List of working devices

### DISCUSSION THREAD

**Discussion by bsiever** [bsiever/microbit-pxt-blehid#21] [General]
If you have been able to pair and get any service working on a device or operating system version that hasn't already been listed, add it here:

**Comment by bsiever** [comment]
My tests:

* Galaxy Tab A (8”, 2019) w/ Android 11
* Microsoft Surface Pro (5th Gen) w/ Windows 10 Pro
* iPhone 11 w/ iOS 15.1
* MacBook Air (M1) w/ 11.4 (Big Sur)

**Comment by themirrazz** [comment]
I have got the keyboard and media services working on a Chromebook running ChromeOS 99. However, Keyboard shortcuts must be sent in a specific way (holding modifiers, send modifiers and rest of shortcut, then releasing modifiers) in order for ChromeOS to register it correctly.

**Reply by bsiever** [reply]
Thanks for sharing!  Just to confirm, the following approach does _not_ work on ChromeOS, right?  https://makecode.microbit.org/_D32FeWiuiCqo (Joining the modifier and key and sending the joined sequence).  

If that's correct, could you share a link to a brief example that _does_ work in case anyone else wants to do this?

**Reply by bsiever** [reply]
I should mention that the example should try to send Control+S (capital S).  Modifying the letter to lower case may be needed in some cases too.

**Comment by bsiever** [comment]
Mouse services seem to work on recent iPads!

**Comment by Mitsuperboy** [comment]
Will this work with a Desktop PC running Windows 10 , and how do I get it to connect? My Desktop has built in WiFi/Bluetooth or do I need to use a USB Bluetooth Dongle of some sort?

**Reply by bsiever** [reply]
It should connect via the regular bluetooth (well, it uses Bluetooth low energy).   I do an example in this video: https://youtu.be/n4J5GN72N_4?t=176 (at around minute 2:56).   I've only been able to test on my one Windows machine, an old Windows Surface 3 with Windows 10.   I hope that helps!  Let me know what you find.

**Reply by bsiever** [reply]
By the way, be sure you include the correct "start service" blocks in your code (as shown in the video).  If you continue to have problems, you can share your code and I can take a look.  Here are the details to create a URL to share: https://support.microbit.org/support/solutions/articles/19000086289-sharing-a-makecode-project .)

**Comment by KillianHaelman** [comment]
Hey i may be 3 years late but i got this working on an acer Chromebook running ChromeOS Version 132.0.6834.222.
The shortcuts work by using one simultaneous keys block for everything, but some shortcuts dont work as they require pressing some keys twice in a row (for example ctrl shift q q logs you out). It'd also be nice to have a way to use the launcher key with the chromebook.

**Reply by bsiever** [reply]
Hi, 

* keys twice:  if back-to-back `sends` don't work (I.e., `send simultaneous keys` with "ctrl shift q" then a `send keys` with "q"), there may be technical limitations that prevent emulating this.  Instead you may be able to change the shortcut to something that can be sent from the micro:bit (doesn't use the additional q).  See: https://support.google.com/chromebook/answer/183101?hl=en . 
* launcher key:  If you know the key code, you can use the `raw scancode` block (in the `... more` part of the `keyboard` toolbox).

**Reply by KillianHaelman** [reply]
Thank you, I'll test this when i get the chance. The reason why i didnt change the shortcut was because it requires you to use the launcher key in the shortcut to prevent interfering with other shortcuts. It should work like this though thanks for responding so quickly:)

**Comment by joselemk** [comment]
I got this running on a ubuntu computer. (with dongle)
